src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -332,117 +270,258 @@ private Vector2 GetRightCornerPointOnMap()
 
-                    if (newStart != start)
+                if (direction is 1 or 3)
+                {
+                    float right = (connectionsToDirection.Count - 1) / 2.0f, left = -right;
+                    foreach (var (connection, endpoint, intermediate, _) in
+                             connectionsToDirection.OrderBy(t => t.Item4))
                     {
-                        start = newStart;
-                        startRegion = drawingAdjacent;
+                        if (connection.Length == 2
+                            || connection[2 * intermediate - endpoint].x > connection[intermediate].x)
+                        {
+                            connection[endpoint].x += lineSeparation * right;
+                            connection[intermediate].x += lineSeparation * right;
+                            right -= 1;
+                        }
+                        else
+                        {
+                            connection[endpoint].x += lineSeparation * left;
+                            connection[intermediate].x += lineSeparation * left;
+                            left += 1;
+                        }
                     }
                 }
-
-                foreach (var drawingAdjacent in adjacent.Adjacent)
+                else
                 {
-                    if (!map.IsSpeciallyDrawnRegion(drawingAdjacent.ID))
-                        continue;
+                    float down = (connectionsToDirection.Count - 1) / 2.0f, up = -down;
+                    foreach (var (connection, endpoint, intermediate, _) in
+                             connectionsToDirection.OrderBy(t => t.Item4))
+                    {
+                        if (connection.Length == 2
+                            || connection[2 * intermediate - endpoint].y > connection[intermediate].y)
+                        {
+                            connection[endpoint].y += lineSeparation * down;
+                            connection[intermediate].y += lineSeparation * down;
+                            down -= 1;
+                        }
+                        else
+                        {
+                            connection[endpoint].y += lineSeparation * up;
+                            connection[intermediate].y += lineSeparation * up;
+                            up += 1;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    ///   Get intersection count and priority for sorting.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Note that priority should be within (-10, 10), and the return value is relative.
+    ///   </para>
+    /// </remarks>
+    private int IntersectionCountWithPriority((Vector2[] Path, int Priority) tuple)
+    {
+        var path = tuple.Path;
+        var priority = tuple.Priority;
+
+        var count = 0;
+
+        for (var i = 1; i < path.Length; ++i)
```

Comment:
```suggestion
        for (int i = 1; i < path.Length; ++i)
```
primitive types when used like this (var count is also the same above) should in my preference say the type. I see this a bunch all over this PR.

Reply:
Emm, rider suggests this. I'll change it back then.